### PR TITLE
docker-compose refactoring

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: deploy/docker-compose.yml [hunk 1]
```diff
@@ -1,53 +1,183 @@
...
+  lighthouse:
+    driver: local
 
 services:
   oracle:
-    build:
-      context: ./../
-      dockerfile: Dockerfile
+    container_name: oracle
+    # build:
```
Comment (tsudmi) [inline deploy/docker-compose.yml:24]: Do we need that commented part here?

File: deploy/docker-compose.yml [hunk 2]
```diff
@@ -1,53 +1,183 @@
...
     restart: always
     entrypoint: ["python"]
     command: ["oracle/oracle/main.py"]
-    env_file:
-      - .env
+    env_file: [".env"]
 
   # keeper:
+  #   container_name: keeper
   #   build:
```
Comment (tsudmi) [inline deploy/docker-compose.yml:35]: Can we use image for keeper here as well?

File: deploy/README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 The deployment directory contains a set of docker-compose files for deploying `oracle` and `keeper` (by default, only `oracle` will be deployed). Oracle requires eth2 node, graph node and ipfs as dependencies, if you do not use cloud solutions, you can deploy self-hosted dependencies:
 
 ```console
-$ docker-compose -f docker-compose.yml -f docker-compose.prysm.yml -f docker-compose.graph.yml -f docker-compose.ipfs.yml up -d
+$ COMPOSE_PROFILES=geth,prysm,graph docker-compose up -d
```
Comment (tsudmi) [inline deploy/README.md:6]: Can we use COMPOSE_PROFILES in `.env`?

File: deploy/docker-compose.yml [hunk 4]
```diff
@@ -1,53 +1,183 @@
...
+    env_file: [".env"]
     volumes:
       - alertmanager:/opt/bitnami/alertmanager/data
       - ./configs/alertmanager.yml:/opt/bitnami/alertmanager/conf/config.yml
-    depends_on:
-      - prometheus
+    depends_on: ["prometheus"]
+
+  graph-node:
+    container_name: graph-node
```
Comment (tsudmi) [inline deploy/docker-compose.yml:65]:
```suggestion
    container_name: graph_node
```

File: deploy/docker-compose.yml [hunk 5]
```diff
@@ -1,53 +1,183 @@
...
+    ports: ["127.0.0.1:9093:9093"]
+    env_file: [".env"]
     volumes:
       - alertmanager:/opt/bitnami/alertmanager/data
       - ./configs/alertmanager.yml:/opt/bitnami/alertmanager/conf/config.yml
-    depends_on:
-      - prometheus
+    depends_on: ["prometheus"]
+
+  graph-node:
```
Comment (tsudmi) [inline deploy/docker-compose.yml:64]:
```suggestion
  graph_node:
```

File: deploy/docker-compose.yml [hunk 6]
```diff
@@ -1,53 +1,183 @@
...
+    command: >
+      /bin/sh -c "until nc -vz graph-node 8020; do echo 'Waiting graph-node'; sleep 2; done
+      && yarn prepare:${NETWORK}
+      && yarn codegen
+      && yarn build
+      && yarn create:local
+      && yarn deploy:local"
+    env_file: [".env"]
+    restart: "no"
+    depends_on: ["graph-node","ipfs"]
```
Comment (tsudmi) [inline deploy/docker-compose.yml:93]:
```suggestion
    depends_on: ["graph_node","ipfs"]
```

File: deploy/docker-compose.yml (hunk 2 above)
Comment (unxnn) [reply deploy/docker-compose.yml:35]: Yes, added

File: deploy/README.md (hunk 3 above)
Comment (unxnn) [reply deploy/README.md:6]: Unfortunately no :(

**Pull request merged**